cpp/src/raft_runtime/solver/lanczos_solver.cuh

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2024-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <raft/sparse/solver/lanczos.cuh>
+#include <raft/spectral/cluster_solvers.cuh>
+#include <raft/spectral/eigen_solvers.cuh>
+#include <raft/spectral/matrix_wrappers.hpp>
+
+#include <raft_runtime/solver/lanczos.hpp>
+
+template <typename IndexType, typename ValueType>
```

Comment:
`raft_runtime` is explicitly for code that's already compiled into RAFT so that users can compile their cpp code against it without needing a CUDA compiler. `raft_runtime` is not for cuda headers. Please move this into `include/raft/solver/detail` and expose a public header just for the public API functions that invoke the solver. Then swap out the spcectral clustering primitive in RAFT to use the new lanczos solver from its public API. Please keep the code well groomed and it will be easy to maintain / update going forward.